history.py

```diff
@@ -0,0 +1,15 @@
+class LastFiveMiddleware
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # One-time configuration and initialization.
+
+    def __call__(self, request):
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.
+
+        response = self.get_response(request)
+
+        # Code to be executed for each request/response after
+        # the view is called.
+
```

Comment:
The session object can only hold standard variables like int or strings. so make sure to convert the product object back into ids